docker/Dockerfile

```diff
@@ -159,11 +159,9 @@ ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-arm64/bin"
 # hadolint ignore=DL3006
 FROM dependencies-dev-platform-$TARGETARCH as dependencies-dev
 
-# Install markdownlint-cli
-RUN npm install --global markdownlint-cli@0.31.1
-
-# Install development-specific dependencies of Nautobot
-RUN poetry install --no-root --no-ansi --extras all
+# Install markdownlint-cli & development-specific dependencies of Nautobot
+RUN npm install --global markdownlint-cli@0.31.1 && \
+    poetry install --no-root --no-ansi --extras all
```

Comment:
To my mind these two tasks are separate enough in their operation that it makes sense to keep them as separate stages. It's not ideal if updating a Python dependency means the Docker build has to reinstall markdown-lint as well because they're combined into a single stage. I'd prefer to keep them separate.

Reply:
As I understand Docker's layer caching strategy it won't matter. Because this line: https://github.com/nautobot/nautobot/pull/1687/files#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddcR106 will have changed, all layers on top will have to get re-run.

I might be misunderstanding this though: https://medium.com/condenastengineering/speedy-builds-with-docker-layer-caching-7ed590ac2fd1

Reply:
Fair point!